packages/components/src/components/link/link.hbs

```diff
@@ -5,9 +5,9 @@
   LICENSE file in the root directory of this source tree.
 -->
 
-<a href="#" class="{{@root.prefix}}--link">Link</a>
-<a class="{{@root.prefix}}--link" aria-disabled="true">Placeholder link</a>
-<p class="{{@root.prefix}}--link--disabled">Disabled Link</p>
+<a href="#" class="{{@root.prefix}}--link{{#if inline}} {{@root.prefix}}--link--inline{{/if}}">Link</a>
+<a class="{{@root.prefix}}--link{{#if inline}} {{@root.prefix}}--link--inline{{/if}}" aria-disabled="true">Placeholder link</a>
+<p class="{{@root.prefix}}--link--disabled{{#if inline}} {{@root.prefix}}--link--inline{{/if}}">Disabled Link</p>
```

Comment:
Is there a reason this is a `p` element here? In the React component, the element remains an `a` element, and has both `bx--link` and `bx--link--disabled` added.

Reply:
It came from #1802, seems that it was for achieving styling requirements. @emyarod Do you have a details? I thought we had a discussion somewhere, but where it was completely slipped my mind (my apologies). My quick searches and following links from #1802 didn't reveal any, but I may be missing something.

Reply:
I think it was to emphasize that simply adding the disabled link class would not be enough to actually disable an anchor. I think an anchor without an `href` could replace the `p` example here

Reply:
I can't find the conversation but we've talked about the disabled links before.  For a11y the demo should be:

```
<span role="link" aria-disabled="true" class="{{@root.prefix}}--link--disabled{{#if inline}} {{@root.prefix}}--link--inline{{/if}}">Disabled Link</span>
```

It's not a paragraph so it shouldn't be using a `p` tag. The `role` tells screen reader users that the intention is that it's a link and the `aria-disabled` informs them that it's disabled